c/auth.c

```diff
@@ -1107,7 +1106,69 @@ int verify_multisig(uint8_t *prefilled_data, const uint8_t *lock_bytes,
     return 0;
 }
 
+static bool require_secp256k1_data(uint8_t algorithm_id) {
+    switch (algorithm_id) {
+        case AuthAlgorithmIdCkb:
+        case AuthAlgorithmIdEthereum:
+        case AuthAlgorithmIdEos:
+        case AuthAlgorithmIdTron:
+        case AuthAlgorithmIdBitcoin:
+        case AuthAlgorithmIdDogecoin:
+        case AuthAlgorithmIdCkbMultisig:
+        case AuthAlgorithmIdSchnorr:
+        case AuthAlgorithmIdLitecoin:
+        case AuthAlgorithmIdRipple:
+            return true;
+        default:
+            return false;
+    }
+    return false;
+}
+
 // dynamic linking entry
+__attribute__((visibility("default"))) int ckb_auth_load_prefilled_data(
+    uint8_t algorithm_id, uint8_t *prefilled_data, size_t *len) {
+    if (require_secp256k1_data(algorithm_id)) {
+        if (prefilled_data == NULL) {
+            if (*len == 0) {
+                *len = CKB_AUTH_RECOMMEND_PREFILLED_LEN;
+                return 0;
+            } else {
+                return ERROR_PREFILLED;
```

Comment:
This code requires `*len` to be 0 when `prefilled_data` is a null pointer, otherwise `ERROR_PREFILLED` is returned. It would be better if we can mention this in the document `auth.md`.

Reply:
mentioned, see auth.md.